repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -4773,25 +4773,55 @@ public List<AtlasVertex> linkBusinessPolicy(String policyId, Set<String> linkGui
 
             updateModificationMetadata(ev);
 
-            cacheDifferentialEntity(ev, existingValues);
+            cacheDifferentialEntity(ev, existingValues, ev.getMultiValuedSetProperty(NON_COMPLIANT_ASSET_POLICY_GUIDS, String.class));
```

Comment:
I think there is a case where policy will be only removed from either ASSET_POLICY_GUIDS or NON_COMPLIANT_ASSET_POLICY_GUIDS, still, we are writing both attributes in audits, this is leads audits. Can you check this please